Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,284 @@
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemas(Dictionary<string, ForgeTree> schemas, JSchema rules, out List<string> errorList)
+        {
+            return Validate(ConvertDictionaryToForgeTreeList(schemas), rules, out errorList);
+        }
+        /// <summary>
+        /// The validate task that validate the schema in the input file path with custom rules in JSchema.
+        /// </summary>
+        /// <param name="path">The path that contains a schema file</param>
+        /// <param name="rules">The rules used to validate input schemas</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaInPath(string path, JSchema rules, out List<string> errorList)
+        {
+            var schema = GetSchemaFromPath(path, out errorList);
+            return CheckConvertErrorAndValidate(rules, ref errorList, schema);
+        }
+        /// <summary>
+        /// The validate task that validate all schemas in a directory with custom rules in JSchema.
+        /// </summary>
+        /// <param name="path">The path that contains a schemas directory</param>
+        /// <param name="rules">The rules used to validate input schemas</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateMultipleSchemasInPath(string path, JSchema rules, out List<string> errorList)
+        {
+            var schemas = GetAllSchemasInDirectory(path, out errorList);
+            return CheckConvertErrorAndValidate(rules, ref errorList, schemas);
+        }
+
+        /// <summary>
+        /// The validate task that check the input schema in string with custom rules in JSchema.
+        /// </summary>
+        /// <param name="schema">The schema to be validated</param>
+        /// <param name="rules">The rules used to validate input schemas</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaString(string schema, JSchema rules, out List<string> errorList)
+        {
+            var schemaList = ConvertStringToJObjectList(schema, out errorList);
+            if (errorList.Count > 0)
+            {
+                return false;
+            }
+            var res = Validate(schemaList, rules, out errorList); 
+            return res;
+        }
+
+        private static List<JObject> ConvertDictionaryToForgeTreeList(Dictionary<string, ForgeTree> schemas)
+        {
+            var schemaList = new List<JObject>();
+            foreach (var item in schemas.Values)
+                schemaList.Add(SerializeForgeTree(item));
```

Comment:
nit style: Always add braces, even if an if/for/while/etc. statement is a single line. Not including braces for a single line can lead to accidental bugs, and is generally not clean code.